### DISCUSSION TITLE
【Solana】CreateAndTransferToken

### DISCUSSION THREAD

**Discussion by LB6264** [trustwallet/wallet-core#1856] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/12064223/146738887-10823e7e-adc9-4cc4-8d16-7f3fb60fd4ee.png)

What does recipientMainAddress mean?

**Comment by LB6264** [comment]
{"jsonrpc":"2.0","error":{"code":-32002,"message":"Transaction simulation failed: Error processing Instruction 0: Provided seeds do not result in a valid address","data":{"accounts":null,"err":{"InstructionError":[0,"InvalidSeeds"]},"logs":["Program ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL invoke [1]","Program log: Error: Associated address does not match seed derivation","Program ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL consumed 10072 of 200000 compute units","Program ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL failed: Provided seeds do not result in a valid address"]}},"id":7}

**Comment by optout21** [comment]
For a main address (which can hold SOL), for a given token, there is a 'token account' that can hold that token.
When transferring a token from a sender to a receiver, there is a sender main address, a sender token address (which holds the foken), a receiver main address, and a receiver token address.
The receiver token address may not exist, in this case it will be created.

The token address can be derived deterministically from the main address and the token mint address, see `TWSolanaAddressDefaultTokenAddress`.

**Reply by LB6264** [reply]
> For a main address (which can hold SOL), for a given token, there is a 'token account' that can hold that token. When transferring a token from a sender to a receiver, there is a sender main address, a sender token address (which holds the foken), a receiver main address, and a receiver token address. The receiver token address may not exist, in this case it will be created.
> 
> The token address can be derived deterministically from the main address and the token mint address, see `TWSolanaAddressDefaultTokenAddress`.

On the transfer interface, I entered the wallet address of a third party. How can I get recipientMainAddress? The senderTokenAddress and recipientTokenAddress parameters are user input. MintAddress can be obtained via getTokenAccountsByOwner, but recipientMainAddress I don't know how to get

**Comment by optout21** [comment]
recipientMainAddress is the wallet address of the intended 3rd party recipient

**Reply by LB6264** [reply]
> recipientMainAddress is the wallet address of the intended 3rd party recipient

If recipientMainAddress is the wallet address of the intended 3rd party recipient, but recipientTokenAddress? Because I did not create an Account under this token, I tried not to send recipientTokenAddress parameters, but the underlying SDK crashed

**Reply by LB6264** [reply]
![image](https://user-images.githubusercontent.com/12064223/146751537-d3e1ce9d-1f11-4faa-9a09-da9fcacf9dd3.png)
The picture above is the demo code of the demo, I don’t know how to get recipientTokenAddress? Because recipientTokenAddress has not yet created an Account corresponding to the token

**Comment by LB6264** [comment, marked as answer]
Thank you for answering the question has been resolved, 
recipientTokenAddress = createAssociatedAccountAddress(to,contractAddress)

**Reply by baahkusi** [reply]
Thanks, this works but it means `CreateAndTransferToken` doesn't work as expected.
